src/main/java/de/tum/in/www1/artemis/repository/ExamRepository.java

```diff
@@ -46,6 +46,10 @@
     @EntityGraph(type = LOAD, attributePaths = { "exerciseGroups", "exerciseGroups.exercises", "registeredUsers", "studentExams" })
     Optional<Exam> findWithExercisesRegisteredUsersStudentExamsById(Long id);
 
+    @EntityGraph(type = LOAD, attributePaths = { "registeredUsers", "exerciseGroups", "exerciseGroups.exercises", "exerciseGroups.exercises.studentParticipations",
+            "exerciseGroups.exercises.studentParticipations.results", "exerciseGroups.exercises.studentParticipations.submissions" })
```

Comment:
I don't think you need "exerciseGroups.exercises.studentParticipations.submissions" here.

Reply:
The DB query is already quite extensive, in particular if we have >1000 students and 6-10 exercises, so we should try to avoid unnecessary data here

Reply:
Changed in 1ebbac866821187dcff834d922b67f799d305262